layer2/lib/test/example_tests.ml

```diff
@@ -0,0 +1,246 @@
+open Tools
+
+let test_empty () = check_lists [] (Mtm.to_list Mtm.empty)
+let test_from_empty_list () = check_lists [] (Mtm.to_list @@ Mtm.from_list [])
+
+let test_from_one_elem_list () =
+  check_lists [ "a", 1 ] (Mtm.to_list @@ Mtm.from_list @@ [ "a", 1 ])
+;;
+
+let test_from_two_elem_list () =
+  check_lists [ "a", 1; "b", 2 ] (Mtm.to_list @@ Mtm.from_list @@ [ "a", 1; "b", 2 ])
+;;
+
+let test_from_three_elem_list () =
+  check_lists
+    [ "a", 1; "b", 2; "c", 3 ]
+    (Mtm.to_list @@ Mtm.from_list @@ [ "b", 2; "a", 1; "c", 3 ])
```

Comment:
`Mtm.to_list` and `Mtm.from_list` should be inverse function for each other to me.
I'd suggest either way:
- `Mtm.from_list` rename as `Mtm.build` and `Mtm.to_list` as `Mtm.flatten`
-  replace `list` with `set`
-  add some commons to indicate `Mtm.to_list @@ Mtm.from_lis` aren't identical

Reply:
I saw it changed to `Seq`  but the name still `to_list` and `from_list` on the top.